google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/application/AppEngineApplicationCreateDialog.java

```diff
@@ -0,0 +1,206 @@
+  private final static String STANDARD_DOCUMENTATION_URL
+      = "https://cloud.google.com/appengine/docs/about-the-standard-environment";
+  private static final String HTML_OPEN_TAG = "<html><font face='sans' size='-1'>";
+  private static final String HTML_CLOSE_TAG = "</font></html>";
+
+  private JPanel panel;
+  private JTextPane instructionsTextPane;
+  private JComboBox<AppEngineLocationSelectorItem> regionComboBox;
+  private JTextPane statusPane;
+  private JTextPane regionDetailPane;
+
+  private final Credential userCredential;
+  private final String gcpProjectId;
+
+  public AppEngineApplicationCreateDialog(@NotNull Component parent, @NotNull String gcpProjectId,
+      @NotNull Credential userCredential) {
+    super(parent, false);
+
+    this.gcpProjectId = gcpProjectId;
+    this.userCredential = userCredential;
+
+    init();
+    setTitle(GctBundle.message("appengine.application.region.select"));
+    refreshLocationsSelector();
+
+    regionComboBox.addItemListener((event) -> {
+      if (event.getStateChange() == ItemEvent.SELECTED) {
+        updateLocationDetailMessage();
+      }
+    });
+
+    regionDetailPane.addHyperlinkListener(new BrowserOpeningHyperLinkListener());
+    instructionsTextPane.addHyperlinkListener(new BrowserOpeningHyperLinkListener());
+    instructionsTextPane.setText(HTML_OPEN_TAG
+        + GctBundle.message("appengine.application.create.instructions") + "<p>"
+        + GctBundle.message("appengine.application.create.documentation",
+        "<a href=\"" + LOCATIONS_DOCUMENTATION_URL + "\">", "</a>") + "</p>" + HTML_CLOSE_TAG);
+  }
+
+  @Override
+  protected void doOKAction() {
+    final Location selectedLocation
+        = ((AppEngineLocationSelectorItem) regionComboBox.getSelectedItem()).getLocation();
+
+    // show loading state
+    setOKActionEnabled(false);
+
+    try {
+      UsageTrackerProvider.getInstance()
+          .trackEvent(GctTracking.APP_ENGINE_APPLICATION_CREATE)
+          .ping();
+
+      // attempt to create the application, and close the dialog if successful
+      ProgressManager.getInstance().runProcessWithProgressSynchronously(() ->
+              AppEngineAdminService.getInstance().createApplication(
+                  selectedLocation.getLocationId(), gcpProjectId, userCredential),
+      GctBundle.message("appengine.application.create.loading",
+              selectedLocation.getLocationId()),
+          true /* cancellable */,
+          ProjectManager.getInstance().getDefaultProject());
+
+      UsageTrackerProvider.getInstance()
+          .trackEvent(GctTracking.APP_ENGINE_APPLICATION_CREATE_SUCCESS)
+          .ping();
+
+      close(OK_EXIT_CODE);
+
+    } catch (IOException e) {
+      trackApplicationCreateFailure();
+      setStatusMessage(GctBundle.message("appengine.application.create.error.transient"), true);
+      setOKActionEnabled(true);
+      return;
+    } catch (GoogleApiException e) {
+      trackApplicationCreateFailure();
```

Comment:
That might be nice, but is there a way to send an error report without throwing an uncaught exception to the user? Ideally I'd like to handle these gracefully and avoid showing the user those red exception popups.